### ISSUE TITLE
noqa get ignored when it is in a block

### ISSUE THREAD

**Issue by braindevices** [ansible-community/ansible-lint#1929]
##### Summary

when the `# noqa` comment is in a nested block, it has no effect any more. This is new in 5.4.0

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.12.2]
ansible-lint --version
WARNING: PATH altered to include /usr/bin
ansible-lint 5.4.0 using ansible 2.12.2

```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Ubuntu20.04

##### STEPS TO REPRODUCE

site.yml
```yaml
---
- name: Setup
  hosts: host
  tasks:
    - debug:  # noqa unnamed-task
        msg: test
    - name: example of multi-level block
      block:
        - name: test var-naming  # noqa var-naming
          set_fact:
            test1: xxx
          vars:
            DISTRIBUTOR_ID: xxx
        - debug:  # noqa unnamed-task
            msg: "test unnamed task"
        - name: 2nd level
          block:
            - name: test var-naming  # noqa var-naming
              set_fact:
                test1: xxx
              vars:
                DISTRIBUTOR_ID: xxx
            - debug:  # noqa unnamed-task
                msg: "test unnamed task"```
```
```console
ansible-lint -v --parseable-severity -c ansible_lint_config site.yml 
WARNING: PATH altered to include /usr/bin
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: site.yml
INFO     Executing syntax check on site.yml (0.34s)
WARNING  Listing 2 violation(s) that are fatal
site.yml:18: [var-naming] [MEDIUM] Task defines variables within 'vars' section that violates variable naming standards
site.yml:23: [unnamed-task] [MEDIUM] All tasks should be named
You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental
  - unnamed-task  # All tasks should be named
Finished with 1 failure(s), 1 warning(s) on 1 files.
```

##### Desired Behavior

Should have no failsure
Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:

- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by braindevices** [comment]
typo

**Comment by braindevices** [comment]
updated the example

**Comment by nre-ableton** [comment]
I also just ran into this bug when updating one of my Ansible projects. This bug along with https://github.com/ansible-community/ansible-lint/issues/1894, is a particularly bad combination that means for some of my projects I’ll be unable to update to 5.4.0.

I know that I can suppress `no-changed-when` globally, but that’s a “nuclear solution” that I’d rather not take.

Hopefully either this bug or #1894 is addressed in the next release.
